### PR TITLE
improve staking cli

### PR DESCRIPTION
**Pull request by zjg555543**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/distribution/client/cli/tx.go [hunk 1]
```diff
@@ -102,20 +108,48 @@ $ %s tx distr withdraw-rewards exvaloper1alq9na49n9yycysh889rl90g9nhe58lcqkfpfg
...
+					delAddr.String(), version.ClientName))
+			}
+			return result
 		},
 	}
 	cmd.Flags().Bool(flagCommission, false, "withdraw validator's commission")
 	return cmd
 }
 
+func transWrapError(wrapErr *common.WrapError, delAddr sdk.AccAddress, valAddr sdk.ValAddress) error {
```
Comment (giskook) [inline x/distribution/client/cli/tx.go:138]: utAck.
cosmos-sdk/types/errors.Wrapf() function can add a description to a error. can we wrap a similar function or put the new wrapError to the  cosmos-sdk/types/errors?

File: x/distribution/client/common/common.go [hunk 2]
```diff
@@ -90,3 +93,78 @@ func WithdrawValidatorRewardsAndCommission(validatorAddr sdk.ValAddress) ([]sdk.
...
+func (e WrapError) Error() string {
+	data, jsonErr := json.Marshal(e)
+	if jsonErr != nil {
+		fmt.Fprintf(os.Stderr, "Trans wrap error, marshal err=%v\n", jsonErr)
+		return ""
+	}
+	return string(data)
+}
+
+func (e *WrapError) SetLog(log string) {
```
Comment (giskook) [inline x/distribution/client/common/common.go]:
```suggestion
func (e *WrapError) setLog(log string) {
```

File: x/distribution/client/common/common.go [hunk 3]
```diff
@@ -90,3 +93,78 @@ func WithdrawValidatorRewardsAndCommission(validatorAddr sdk.ValAddress) ([]sdk.
...
+
+type WrapError struct {
+	Code      uint32 `json:"code"`
+	Log       string `json:"log"`
+	Codespace string `json:"codespace"`
+	Changed   bool   `json:"-"`
+	RawError  error  `json:"-"`
+}
+
+func (e WrapError) Error() string {
```
Comment (giskook) [inline x/distribution/client/common/common.go]:
```suggestion
func (e *WrapError) Error() string {
```

File: x/distribution/client/cli/tx.go (hunk 1 above)
Comment (zjg555543) [reply x/distribution/client/cli/tx.go:138]: cosmos-sdk/types/errors is **desc**,but wrap is **log**
type Error struct {
	codespace string
	code      uint32
	desc      string
}

type WrapError struct {
	Code      uint32 `json:"code"`
	Log       string `json:"log"`
	Codespace string `json:"codespace"`
}

**Review by LeoGuo621 — APPROVED** [review]

**Review by giskook — APPROVED** [review]

**Pull request merged**